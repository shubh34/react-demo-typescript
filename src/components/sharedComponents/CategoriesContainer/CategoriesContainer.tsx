import './CategoriesContainer.css';
import React from 'react';

import { Icon } from 'semantic-ui-react';
import Category from '../Category/Category';

interface categoriesProps {
	categories: string[];
	hasTagIcon?: boolean;
	hasDeleteIcon?: boolean;
	onCategoryClick?: Function;
}

const CategoriesContainer: React.FC<categoriesProps> = props => {
	const { categories, hasTagIcon, hasDeleteIcon = false, onCategoryClick } = props;
	return (
		<div className='categories_container'>
			{hasTagIcon && <Icon name='tags' />}
			<div className='categories_button_group'>
				{categories.map(category => (
					<Category
						key={category}
						name={category}
						hasDeleteIcon={hasDeleteIcon}
						onCategoryClick={onCategoryClick}
					/>
				))}
			</div>
		</div>
	);
};

export default CategoriesContainer;
