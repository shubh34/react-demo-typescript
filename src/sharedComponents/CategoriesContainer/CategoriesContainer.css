.categories_container {
    padding: 16px 0px;
}
.categories_button_group {
    display: inline;
}