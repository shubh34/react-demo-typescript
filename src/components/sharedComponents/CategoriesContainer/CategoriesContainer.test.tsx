import React from 'react';
import { shallow } from 'enzyme';
import CategoriesContainer from './CategoriesContainer';

const setUpContainer = (setUpProps = {}, container= shallow) => {
    const props =  {
        categories: ['Food', 'Travel'],
        ...setUpProps
    }
    const wrapper = shallow(<CategoriesContainer {...props} />);
    return {
        props,
        wrapper,
    }
}
describe('CategoriesContainer', () => {
    it('should Render CategoriesContainer with Category Icon', () => {
        const wrapper = setUpContainer({ hasTagIcon: true });
        expect(wrapper).toMatchSnapshot();
    });
    it('should render CategoriesContainer without Category Icon', () => {
        const wrapper = setUpContainer({ hasTagIcon: false });
        expect(wrapper).toMatchSnapshot();
    });
    it('should render category button  with deleteIcon', () => {
        const wrapper = setUpContainer({ hasDeleteIcon: true });
        expect(wrapper).toMatchSnapshot();
    });
})