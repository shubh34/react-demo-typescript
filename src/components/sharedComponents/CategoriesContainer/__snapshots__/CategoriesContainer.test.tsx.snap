// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoriesContainer should Render CategoriesContainer with Category Icon 1`] = `
Object {
  "props": Object {
    "categories": Array [
      "Food",
      "Travel",
    ],
    "hasTagIcon": true,
  },
  "wrapper": <div
    className="categories_container"
  >
    <Icon
      as="i"
      name="tags"
    />
    <div
      className="categories_button_group"
    >
      <Category
        hasDeleteIcon={false}
        key="Food"
        name="Food"
      />
      <Category
        hasDeleteIcon={false}
        key="Travel"
        name="Travel"
      />
    </div>
  </div>,
}
`;

exports[`CategoriesContainer should render CategoriesContainer without Category Icon 1`] = `
Object {
  "props": Object {
    "categories": Array [
      "Food",
      "Travel",
    ],
    "hasTagIcon": false,
  },
  "wrapper": <div
    className="categories_container"
  >
    <div
      className="categories_button_group"
    >
      <Category
        hasDeleteIcon={false}
        key="Food"
        name="Food"
      />
      <Category
        hasDeleteIcon={false}
        key="Travel"
        name="Travel"
      />
    </div>
  </div>,
}
`;

exports[`CategoriesContainer should render category button  with deleteIcon 1`] = `
Object {
  "props": Object {
    "categories": Array [
      "Food",
      "Travel",
    ],
    "hasDeleteIcon": true,
  },
  "wrapper": <div
    className="categories_container"
  >
    <div
      className="categories_button_group"
    >
      <Category
        hasDeleteIcon={true}
        key="Food"
        name="Food"
      />
      <Category
        hasDeleteIcon={true}
        key="Travel"
        name="Travel"
      />
    </div>
  </div>,
}
`;
